docs/hello_nextflow/02_hello_channels.md

```diff
@@ -0,0 +1,880 @@
+
+```console title="Output" linenums="1"
+ N E X T F L O W   ~  version 24.10.0
+
+Launching `hello-channels.nf` [tiny_elion] DSL2 - revision: 1d834f23d2
+
+executor >  local (3)
+[8e/bb08f3] sayHello (2) | 3 of 3 ✔
+Before flatten: [Hello, Bonjour, Holà]
+After flatten: Hello
+After flatten: Bonjour
+After flatten: Holà
+```
+
+You see that we get a single `Before flatten:` statement because at that point the channel contains one item, the original array.
+Then we get three separate `After flatten:` statements, one for each greeting, which are now individual items in the channel.
+
+Importantly, this means each item can now be processed separately by the workflow.
+
+!!! tip
+
+    You should delete or comment out the `view()` statements before moving on.
+
+    ```groovy title="hello-channels.nf" linenums="31"
+    // create a channel for inputs
+    greeting_ch = Channel.of(greetings_array)
+                         .flatten()
+    ```
+
+    We left them in the `hello-channels-3.nf` solution file for reference purposes.
+
+### Takeaway
+
+You know how to use an operator like `flatten()` to transform the contents of a channel, and how to use the `view()` directive to inspect channel contents before and after applying an operator.
+
+### What's next?
+
+Learn how to make the workflow take a file as its source of input values.
+
+---
+
+## 4. Use an operator to parse input values from a CSV file
+
+It's often the case that, when we want to run on multiple inputs, the input values are contained in a file.
+As an example, we prepared a CSV file called `greetings.csv` containing several greetings, one on each line (like a column of data).
+
+```csv title="greetings.csv" linenums="1"
+Hello
+Bonjour
+Holà
+```
+
+So now we need to modify our workflow to read in the values from a file like that.
+
+### 4.1. Modify the script to expect a CSV file as the source of greetings
+
+To get started, we're going to need to make two key changes to the script:
+
+- Switch the input parameter to point to the CSV file
+- Switch to a channel factory designed to handle a file
+
+#### 4.1.1. Switch the input parameter to point to the CSV file
+
+Remember the `params.greeting` parameter we set up in Part1?
+We're going to update it to point to the CSV file containing our greetings.
+
+In the workflow block, make the following code change:
+
+_Before:_
+
+```groovy title="hello-channels.nf" linenums="25"
+/*
+ * Pipeline parameters
+ */
+params.greeting = ['Hello','Bonjour','Holà']
+```
+
+_After:_
+
+```groovy title="hello-channels.nf" linenums="25"
+/*
+ * Pipeline parameters
+ */
+params.greeting = 'greetings.csv'
+```
+
+#### 4.1.2. Switch to a channel factory designed to handle a file
+
+Since we now want to use a file instead of simple values as the input, we can't use the `Channel.of()` channel factory from before.
```

Comment:
```suggestion
Since we now want to use a file instead of simple strings as the input, we can't use the `Channel.of()` channel factory from before.
```